packages/cdk8s-plus/src/service.ts

```diff
@@ -272,6 +272,37 @@ export class ServiceSpecDefinition {
     return [...this._ports];
   }
 
+  /**
+   * Exposes a deployment through this service.
+   *
+   * Requests will be routed to the port exposed by the first container in the
+   * deployment's pods. The label `cdk8s.deployment` will be added to pods in
+   * the deployment and used to select pods by this service.
+   *
+   * @param deployment The deployment to expose
+   * @param port The external port
+   */
+  public addDeployment(deployment: Deployment, port: number) {
```

Comment:
I'm curious why you decided to implement it like this. Also - is there meaning to adding 2 deployments to a service? wouldn't that cause the service selector to include both deployment labels and break the matching?

Reply:
I wanted an API that allows me to associate a deployment to an existing service (instead of `expose()` which define a new service). The semantic `addXxx` is like saying "add a deployment to the definition of this service".

However, you are right that adding multiple deployments will basically not work. I guess we can fail `addDeployment()` is there are already selectors defined for this service. What do you think?

Reply:
> The semantic addXxx is like saying "add a deployment to the definition of this service"

Problem is that a deployment doesn't really appear in the spec of a service, so i'm not sure its the best semantic. 
I wonder if maybe `service.serve` might be better. i.e, "this service will serve this deployment" (failing when its called twice) - And change the current `serve` to `bind`? 

Granted it doesn't follow the "add to spec definition" semantic but i dont feel like we have clear convention here..

Reply:
Btw - we can also just have the `deployment.expose` method accept an optional service

Reply:
We are getting rid of specs anyway, no? I can move this method from `ServiceSpec` to `Service`. Would that make more sense?

Reply:
done